apps/examples/artik_demo/artik_demo.c

```diff
@@ -747,26 +747,20 @@ int send_data_to_artik(int *data)
 		if (ret < 0) {
 			/* ERROR: Connection Failed */
 			ndbg("ERROR: connect failed: %d\n", ret);
-			close(sockfd);
-			free(client_tls);
-			return ERROR;
+			goto error_fd_tls;
 		}
 
 		client_tls->client_fd = sockfd;
 
 		if (webclient_tls_init(client_tls, &ssl_config)) {
 			/* ERROR: Fail to client tls init */
 			ndbg("Fail to client tls init\n");
-			close(sockfd);
-			free(client_tls);
-			return ERROR;
+			goto error_fd_tls;
 		} else {
 			if (wget_tls_handshake(client_tls)) {
 				/* ERROR: Fail to client tls handshake */
 				ndbg("Fail to client tls handshake\n");
-				close(sockfd);
-				free(client_tls);
-				return ERROR;
+				goto error_fd_tls;
```

Comment:
In this case, tls is initialized. Don't we need to release tls?

Reply:
wget_tls_release(client_tls);
wget_tls_ssl_release(client_tls); 
In case of an error, above two functions are already called in wget_tls_handshake before returning. but client_tls memory is not freed. Hence only free(client_tls) is added.